source/guides/maintenance/custom-app-container.rst

```diff
@@ -0,0 +1,402 @@
+the :command:`swupd_args="-m <CL_VERSION>"` argument. For example, if the
+desired |CL| version is 30550, then the specific command would be:
+
+
+.. code-block:: bash
+
+	$ docker build --no-cache=true --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t clearlinux/machine-learning:30550 --build-arg swupd_args="-m 30550 " machine-learning/
+
+Multi-stage Clear Linux OS Dockerfiles
+--------------------------------------
+
+As mentioned above, all of the higher level layers of multi-stage |CL|
+Dockerfiles inherit the |CL| version from this base layer. So, this requires
+us to first rebuild the base layer against the target OS version.
+
+This can be done by simply adding an extra :command:`swupd_args` argument to
+the command line. For example, if the desired OS version is 30550, then the
+specific command would be:
+
+.. code-block:: bash
+
+	$ docker build --no-cache=true --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t clearlinux/os-core:30550 --build-arg swupd_args="-m 30550" os-core/
+
+Next, you must change the target Dockerfile, because the official version is
+set to always build off the latest base image (that is, clearlinux/os-core:latest).
+
+To demonstrate this, a diff of a modified clearlinux/redis Dockerfile is
+provided below to use the previously built clearlinux/os-core:30550 container
+image.
+
+.. code-block:: bash
+
+	diff --git a/redis/Dockerfile b/redis/Dockerfile
+	index af977cb..d666e6e 100644
+	--- a/redis/Dockerfile
+	+++ b/redis/Dockerfile
+	@@ -7,7 +7,7 @@ RUN swupd update --no-boot-update $swupd_args
+
+	 # Grab os-release info from the minimal base image so
+	 # that the new content matches the exact OS version
+	-COPY --from=clearlinux/os-core:latest /usr/lib/os-release /
+	+COPY --from=clearlinux/os-core:30550 /usr/lib/os-release /
+
+	 # Install additional content in a target directory
+	 # using the os version from the minimal base
+	@@ -26,7 +26,7 @@ COPY --from=clearlinux/os-core:latest / /os_core_install/
+	 RUN cd / && \
+	     find os_core_install | sed -e 's/os_core_install/install_root/' | xargs rm -d &> /dev/null || true
+
+	-FROM clearlinux/os-core:latest
+	+FROM clearlinux/os-core:30550
+	 MAINTAINER qi.zheng@intel.com
+
+	 COPY --from=builder /install_root /
+
+
+	$ docker build --no-cache=true --build-arg http_proxy=$http_proxy --build-arg https_proxy=$https_proxy -t clearlinux/redis:30550 redis/
+
+
+Some of the |CL| containers use more than one base layer. For
+example, the clearlinux/cgit container is built off clearlinux/http:latest
+(that is, it contains a :command:`FROM clearlinux/httpd:latest` line in its Dockerfile),
+which in turn is built off the clearlinux/os-core container. This means all
+three layers need to be locked down against the desired OS version.
+
+Similar to the previous examples, we can lock down both of these containers to
+OS version 30550 by making the following changes to the Dockerfile:
+
+.. code-block:: bash

```

Comment:
This one big codeblock should probably be split into 4 different code blocks. 

1. diff
2. docker build
3. diff again
4. docker build again.